include/poggles/gltypes.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <concepts>
+
+#include <glad/glad.h>
+
+#include "poggles/poggles_export.hpp"
+
+template<typename T>
+concept GLuintValued = requires(T a) {
+                         {
+                           a.id()
+                           } -> std::convertible_to<GLuint>;
+                       };
+
+template<typename T>
+class gltype
+{
+protected:
+  T m_id;
+  // Protected constructor so it can't be created directly (Should only be
+  // created using handle class)
+  gltype(T v)
```

Comment:
Should this constructor be marked explicit? So that we don't accidentally implicitly convert to this type?

Reply:
Oh good point, since it's protected I didn't think too much about it, but I can't think of a good reason for it not to be explicit.